### PR TITLE
Merge RNN Workflows for Simulation Experiments and MIMIC

### PR DESCRIPTION
**Pull request by rathp**
# Simulated Data Experiments for RNN validation
## 2-state model 
- Time series data is simulated from a simple 2 state Markov model where samples at each time point are drawn from a background state or state A(both gaussian), such that samples drawn from each state are clearly distinguishable visually. The sequences where a state transition occurs from background to state A are labeled 1.

-  Usage :
-- Fixing parameter with single job :
`>> cd src/rnn/`
`>> $simulation_data=2-state bash do_rnn_validation.slurm`
-- Multiple jobs parallelizing across parameters : 
`>>  bash launch_rnn_on_simulated_data.sh submit`

## 3-state model 
- Time series data is simulated from a simple 3 state Markov model where samples at each time point are drawn from a background state or state A or state B(both gaussian), such that samples drawn from each state are clearly distinguishable visually. The sequences where a state transition occurs from state A to state B are labeled 1. This is a harder problem because the visual distinction between label 1 sequences and label 2 sequences are not as visually distinguishable as the 2 state model.

-  Usage :
-- Fixing parameter with single job :
`>> cd src/rnn/`
`>> $simulation_data=3-state bash do_rnn_validation.slurm`
-- Multiple jobs parallelizing across parameters : 
`>> bash launch_rnn_on_simulated_data.sh submit`

## MIMIC
- RNN implementation for predicting inhospital mortality. Currently the RNN achieves an AUROC of ~0.7 on validation set but does not ouptperform logistic regression which achieves AUROC ~0.8. 

-  Usage :
-- Fixing parameter with single job :
`>> cd src/rnn/`
`>> $n_epochs=10000 $hidden_layer_sizes=32 $lr=.1 $batch_size=3000 $dropout=0.1 bash do_mortality_prediction_on_mimic_with_rnn.slurm`
-- Multiple jobs parallelizing across parameters : 
`>> bash launch_rnn_on_mimic.sh submit`

## Results and Exact Hyper-Parameter Settings 
Please refer to the following document on the current RNN results and hyper-parameter settings for the simulated experiments and mortality prediction for MIMIC:
[https://docs.google.com/document/d/1u3Snwvh9yMNKXJrFHNfY3txxaqEmGtyVMz_1QM11jQ8/edit](url)

### CODE REVIEW

**Comment by michaelchughes** [comment]
I've updated this significantly to integrate my recent updates to the new Snakemake workflows.

Still TODO: update to do RNN style learning from variable-length time series in this workflow

# Big changes

For each dataset, you can now do

```
cd scripts/{dataset_name}

## build standardized dataset
cd standardize_dataset && snakemake --cores 2 all

## build collapsed dataset for a particular binary prediction task
cd predictions_collapsed
snakemake -s make_collapsed_dataset_and_split_train_test.smk --cores 2 all
snakemake -s train_logistic_regression.smk --cores 2 all

```

**Comment by michaelchughes** [comment]
Merging now, this is quality enough to move to the main branch.
